TPC/TPCcalib/AliTPCcalibAlign.cxx

```diff
@@ -715,10 +715,13 @@ void  AliTPCcalibAlign::ExportTrackPoints(AliVEvent *event){
     AliTPCseed *seed0 = 0,*seed1=0;
     AliTPCseed tpcSeed0, tpcSeed1;
     //
+    Int_t nFriendTracks = Vfriend->GetNumberOfTracks();
+    if (index0 > nFriendTracks) continue; // most likely, we didn't save the friends for this event
```

Comment:
ciao, wouldn't a better check be: if (ntracks != nFriendTracks) ? Or can we have more tracks than friend tracks while having a non-zero number of friend tracks?

Reply:
Hello,

Probably you are right, I am quite sure that @shahor02 could reply with 100% certainty :) 

Cheers,

Chiara

Reply:
@mkrzewic : since LHC15o we are storing maximum ````AliReconstruction::fMaxFriendTracks (def=3000)```` tracks, see https://alice.its.cern.ch/jira/browse/ALIROOT-6411?focusedCommentId=161468&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-161468 for details. So, in high mult events we can have ````0<Nfriends<Ntracks````

Reply:
Hi Ruben, thanks. Then, if I understand it correctly the check should be `if (esdTrack->GetFriendNotStored())`, right?

Reply:
...or better yet, just check for null pointer as the friends are attached to tracks based on criteria other than just the index.

Reply:
The pointer is already checked but there was an error - now I don't recall the details, but I can reproduce it back.

Reply:
Actually, the correct way (compatible with pre-2015 and after data) to get the friends is AliESDtrack->GetFriendTrack(), which will return null if this particular friend was not stored.

Reply:
exactly, that is the problem. for code that also needs to run online on the HLT this would not work.